sys/include/net/nanocoap/link_format.h

```diff
@@ -45,25 +45,29 @@ typedef int (*coap_link_format_handler_t)(char *entry, void *ctx);
  * @param[in]   path    path of the resource
  * @param[in]   cb      Callback to execute for each resource entry
  * @param[in]   arg     Optional callback argument
+ * @param[out]  response Response buffer
  *
  * @returns     0 on success
  * @returns     <0 on error
  */
 int nanocoap_link_format_get(nanocoap_sock_t *sock, const char *path,
-                             coap_link_format_handler_t cb, void *arg);
+                             coap_link_format_handler_t cb, void *arg,
+                             char response[CONFIG_NANOCOAP_QS_MAX]);
```

Comment:
Since we are now supplying the buffer, we aren't bound to that weird `CONFIG_NANOCOAP_QS_MAX` anymore

```suggestion
                             char *work_buf, size_t work_buf_len);
```